haas/client/base.py

```diff
@@ -0,0 +1,38 @@
+""" This module implements the HaaS client library. """
+
+import requests
+import os
+import json
+from haas.client import errors
+from urlparse import urljoin
+
+
+class ClientBase(object):
+    """ Main class which contains all the methods to
+    -- ensure input complies to API requisites
+    -- generates correct format for server API on behalf of the client
+    -- parses output from received from the server.
+    In case of errors recieved from server, it will generate appropriate
+    appropriate message.
+    """
+
+    def __init__(self, endpoint=None, sess=None):
+        """ Initialize an instance of the library with following parameters.
+        endpoint: stands for the http endpoint eg. endpoint=http://127.0.0.1
+        user: username as which you wish to connect to HaaS
+        password: password for the 'user' as decribed above.
+        Currently all this information is fetched from the user's environment.
+        """
+        self.endpoint = endpoint
+        self.s = sess
+
+        if None in [self.endpoint, self.s]:
+            raise LookupError(
+                    "Insufficient attributes to establish "
+                    "connection with HaaS server")
```

Comment:
This message gets raised when people start accessing HIL without initializing the client library. That is not having the requisite environment variable. Without the LookupError, it throws lot of lines on the output and user is left to figure out what went wrong. I felt this is going to be very common issue with new people getting acquainted with HIL, hence a good error messsage may save some time. As to its opaque nature, I will try to make it more informative.